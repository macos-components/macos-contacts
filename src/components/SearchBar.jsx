import React, { Component } from 'react';
import PropTypes from 'prop-types';

import PrimaryLink from 'components/PrimaryLink';

import colors from 'style/colors';
import sizing from 'style/sizing';
import ordering from 'style/ordering';

const containerStyle = {
  ...sizing.flex,
  padding: '10px 15px 10px',
  zIndex: ordering.searchBarZindex,
  position: 'relative',
  backgroundColor: colors.searchBarBackground
};

const itemStyle = {
  ...sizing.flexBox,
  borderRadius: '3px',
  border: '1px solid',
  borderColor: colors.inputBorder,
  padding: sizing.inputPadding
};

const inActiveItemStyle = {
  ...itemStyle,
  textAlign: 'center'
};

const activeItemStyle = {
  ...itemStyle,
  textAlign: 'left',
  outline: 'none'
};

const linkStyle = {
  ...sizing.flexBox,
  marginLeft: '10px'
};

class SearchBar extends Component {
  onFocus = () => {
    if (this.props.onActiveChange) {
      this.props.onActiveChange(true);
    }
  }

  onBlur = () => {
    if (this.props.onActiveChange) {
      this.props.onActiveChange(false);
    }
  }

  render() {
    return (
      <div className="search-bar" style={containerStyle}>
        <input
          style={this.props.isActive ? activeItemStyle : inActiveItemStyle}
          type="text"
          placeholder={this.props.placeholder}
          onFocus={this.onFocus}
          onBlur={this.onBlur}
          onKeyUp={this.props.onKeyUp || null}
        />
        { this.props.isActive ? (
          <div>
            <PrimaryLink label="Cancel" style={linkStyle} />
          </div>
        ) : null }
      </div>
    );
  }
}

SearchBar.propTypes = {
  onActiveChange: PropTypes.func,
  isActive: PropTypes.bool,
  placeholder: PropTypes.string,
  onKeyUp: PropTypes.func
};

SearchBar.defaultProps = {
  onActiveChange: null,
  isActive: false,
  placeholder: 'Search',
  onKeyUp: null
};

export default SearchBar;
