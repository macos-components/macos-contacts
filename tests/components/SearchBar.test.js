import React, { Component } from 'react';
import SearchBar from 'components/SearchBar';
import renderer from 'react-test-renderer';

test('Renders => components/SearchBar', () => {
  const component = renderer.create(
    <SearchBar />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
