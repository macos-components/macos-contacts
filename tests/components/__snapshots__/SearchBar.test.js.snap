// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders => components/SearchBar 1`] = `
<div
  className="search-bar"
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#cccccc",
      "display": "flex",
      "justifyContent": "space-between",
      "padding": "10px 15px 10px",
      "position": "relative",
      "zIndex": "101",
    }
  }
>
  <input
    onBlur={[Function]}
    onFocus={[Function]}
    onKeyUp={null}
    placeholder="Search"
    style={
      Object {
        "border": "1px solid",
        "borderColor": "#dddddd",
        "borderRadius": "3px",
        "flexGrow": "1",
        "padding": "5px",
        "textAlign": "center",
      }
    }
    type="text"
  />
</div>
`;
